convert2rhel/actions/is_loaded_kernel_latest.py

```diff
@@ -0,0 +1,171 @@
+        # meaning that the tool will fetch only the latest kernels available for
+        # that EUS version, and not the most updated version from other newer
+        # versions.
+        repoquery_output, return_code = run_subprocess(cmd, print_output=False)
+        if return_code != 0:
+            logger.debug("Got the following output: %s", repoquery_output)
+            logger.warning(
+                "Couldn't fetch the list of the most recent kernels available in "
+                "the repositories. Skipping the loaded kernel check."
+            )
+            return
+
+        packages = []
+        # We are expecting a repoquery output to be similar to this:
+        #   C2R     1671212820      3.10.0-1160.81.1.el7    updates
+        # We need the `C2R` identifier to be present on the line so we can know for
+        # sure that the line we are working with is a line that contains
+        # relevant repoquery information to our check, otherwise, we just log the
+        # information as debug and do nothing with it.
+        for line in repoquery_output.split("\n"):
+            if line.strip() and "C2R" in line:
+                _, build_time, latest_kernel, repoid = tuple(str(line).split("\t"))
+                packages.append((build_time, latest_kernel, repoid))
+            else:
+                # Mainly for debugging purposes to see what is happening if we got
+                # anything else that does not have the C2R identifier at the start
+                # of the line.
+                logger.debug("Got a line without the C2R identifier: %s" % line)
+
+        # If we don't have any packages, then something went wrong, we need to
+        # decide whether to bail out or output a warning (only if the user used the
+        # special environment variable for it.
+        if not packages:
+            unsupported_skip = os.environ.get("CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK", None)
+            if not unsupported_skip:
+                self.set_result(
+                    status="ERROR",
+                    error_id="KERNEL_CURRENCY_CHECK_FAIL",
+                    message=(
+                        "Could not find any %s from repositories to compare against the loaded kernel.\n"
+                        "Please, check if you have any vendor repositories enabled to proceed with the conversion.\n"
+                        "If you wish to ignore this message, set the environment variable "
+                        "'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' to 1." % package_to_check
+                    ),
+                )
+                return
+
+            logger.warning(
+                "Detected 'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' environment variable, we will skip "
+                "the %s comparison.\n"
+                "Beware, this could leave your system in a broken state. " % package_to_check
+            )
+            return
+
```

Comment:
A small thing you can do regarding the pylint error is to eliminate this return and have something like

```suggestion
    else:
```